kubectl/private-network-clique/README.md

```diff
@@ -0,0 +1,114 @@
+
+## Flow of the process:
+- Create private/public keys for the validators & update the secrets/validator-keys-secret.yaml with the validator private keys
+- Update the configmap/configmap.yml with the public keys & genesis file
+- Update the number of nodes you would like in deployments/node-deployment.yaml
+- Run kubectl
+- Monitoring via prometheus & grafana is also setup up in a separate *monitoring* namespace and exposed via NodePort services (ports 30090, 30030 respectively)
+- Credentials for grafana are admin:admin. When grafana loads up select the "besu Dashboard"
+
+## Overview of Setup
+![Image ibft](../../images/ibft.png)
+
+## NOTE:
+1. validators1 and 2 serve as bootnodes as well. Adjust according to your needs
+2. If you add more validators in past the initial setup, they need to be voted in to be validators i.e they will serve as normal nodes and not validators until they've been voted in.
+
+#### 1. Boot nodes private keys
+Create private/public keys for the validators using nodejs script. The private keys are put into secrets and the public keys go into a configmap to get the bootnode enode address easily
+Repeat this process for as many validators as you would like to provision i.e keys and replicate the deployment & service
+
+```bash
+npm install besu-clique-config
+besu-clique-config --config-file=cliqueSetup/cliqueConfigFile.json --to=cliqueSetup/networkFiles --private-key-file-name=key
+sudo chown -R $USER:$USER ./cliqueSetup
+mv ./cliqueSetup/networkFiles/genesis.json ./cliqueSetup/
+```
+
+Update the secrets/validator-key-secret.yaml with the private keys. The private keys are put into secrets and the public keys go into a configmap that other nodes use to create the enode address
+Update the configmap/configmap.yaml with the public keys
+**Note:** Please remove the '0x' prefix of the public keys
+
+#### 2. Genesis.json
+Copy the genesis.json file and copy its contents into the configmap/configmap as shown
+
```

Comment:
Also worth adding something here with something along the lines of `Update the extra data with addresses that match you validators by concatenating them`

Reply:
@joshuafernandes  `besu-clique-config` will take care of replacing creating the correct `extraData` in the generated genesis file. Basically, there is no need to add it manually.

Reply:
:+1: